tests/e2e/utils/workspace/TestWorkspaceUtil.ts

```diff
@@ -46,46 +57,217 @@ export class TestWorkspaceUtil implements ITestWorkspaceUtil {
+        if ( keycloakUrl.substr(7, 4).includes('che')) {
+            const keycloakAuthSuffix = '/auth/realms/che/protocol/openid-connect/token';
+            keycloakUrl = keycloakUrl.replace('che', 'keycloak') + keycloakAuthSuffix;
+            params = {
+                client_id: 'che-public',
+                username: TestConstants.TS_SELENIUM_USERNAME,
+                password: TestConstants.TS_SELENIUM_PASSWORD,
+                grant_type: 'password'
+            };
+        } else {
+            const keycloakAuthSuffix = '/auth/realms/codeready/protocol/openid-connect/token';
+            keycloakUrl = keycloakUrl.replace('codeready', 'keycloak') + keycloakAuthSuffix;
+            params = {
+                client_id: 'codeready-public',
+                username: TestConstants.TS_SELENIUM_USERNAME,
+                password: TestConstants.TS_SELENIUM_PASSWORD,
+                grant_type: 'password'
+            };
+        }
+
+        try {
+            const responseToObtainBearerToken = await axios.post(keycloakUrl, querystring.stringify(params));
+            return responseToObtainBearerToken.data.access_token;
+        } catch (err) {
+            console.log(`Can not get bearer token. URL used: ${keycloakUrl}`);
+            throw err;
+        }
+
+    }
+
+    public async stopWorkspaceById(id: string) {
+        const stopWorkspaceApiUrl: string = `${this.workspaceApiUrl}/${id}/runtime`;
+        try {
+            const stopWorkspaceResponse: rm.IRestResponse<any> = await this.processRequest(RequestType.DELETE, stopWorkspaceApiUrl);
+
+            // response code 204: "No Content" expected
+            if (stopWorkspaceResponse.statusCode !== 204) {
+                throw new Error(`Can not stop workspace. Code: ${stopWorkspaceResponse.statusCode} Result: ${stopWorkspaceResponse.result}`);
+            }
+        } catch (err) {
+            console.log(`Stopping workspace failed. URL used: ${stopWorkspaceApiUrl}`);
+            throw err;
+        }
+
+    }
+
+    public async cleanUpAllWorkspaces() {
+        let listOfRunningWorkspaces: Array<string> = await this.getIdOfRunningWorkspaces();
+        for (const entry of listOfRunningWorkspaces) {
+            await this.stopWorkspaceById(entry);
+        }
+
+        let listAllWorkspaces: Array<string> = await this.getListOfWorkspaceId();
+
+        for (const entry of listAllWorkspaces) {
+            this.removeWorkspaceById(entry);
+        }
+
+    }
+
+    removeWorkspace(namespace: string, workspaceId: string): void {
         throw new Error('Method not implemented.');
     }
 
-    stopWorkspaceById(id: string): void {
+    stopWorkspace(namespace: string, workspaceId: string): void {
         throw new Error('Method not implemented.');
     }
+
+    async processRequest(reqType: RequestType, url: string) {
+        var response: rm.IRestResponse<any>;
+        if (TestConstants.TS_SELENIUM_MULTIUSER === true) {
```

Comment:
Why var?

Reply:
Good point, I'll go through my code and change `var` to `let`.

Reply:
I've found `var` used in Workspace.ts class to so I change that to `let` as a part of this refactoring too.